### PR TITLE
add distillation forward output class to distillation pipeline.

### PR DESCRIPTION
**Pull request by entrpn**
# Description

Addresses comments from https://github.com/AI-Hypercomputer/maxtext/pull/3218.
- Cleaner class definitions for output values of the forward function for student/teacher.
- Fixes eval loss function.

# Tests

- tests.unit.train_distill_test

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: tests/unit/train_distill_test.py [hunk 1]
```diff
@@ -169,15 +169,21 @@ def test_monitored_strategy(self):
 
     # Dummy inputs (batch=1, seq=2, vocab=4)
     # Note: Shapes must align for broadcasting
-    student_logits = (jnp.array([[[1.0, 0.0, 0.0, 0.0], [0.0, 1.0, 0.0, 0.0]]]) * 10, jnp.ones((32, 1, 1, 8)))
-    teacher_logits = (jnp.array([[[1.0, 0.0, 0.0, 0.0], [0.0, 1.0, 0.0, 0.0]]]) * 10, jnp.ones((32, 1, 1, 8)))
+    student_output = distillation_utils.DistillationForwardOutput(
+        logits=jnp.array([[[1.0, 0.0, 0.0, 0.0], [0.0, 1.0, 0.0, 0.0]]]) * 10,
+        out_projection_activations=np.ones((32, 1, 1, 8)),
```
Comment (gagika) [inline tests/unit/train_distill_test.py]: nit: np.ones ->jnp.ones

**Review by gagika — APPROVED** [review]

File: tests/unit/train_distill_test.py (hunk 1 above)
Comment (entrpn) [reply tests/unit/train_distill_test.py]: done

**Review by vlad-karp — APPROVED** [review]
LGTM
